server/app/services/applicant/ApplicantService.java

```diff
@@ -82,24 +84,64 @@ CompletionStage<ReadOnlyApplicantProgramService> getReadOnlyApplicantProgramServ
   ReadOnlyApplicantProgramService getReadOnlyApplicantProgramService(
       Application application, ProgramDefinition programDefinition);
 
-  /**
-   * Return all programs that are appropriate to serve to an applicant - which is any active program
-   * that is public and any program where they have an application in the draft stage.
-   *
-   * <p>The programs do not have question definitions loaded into its program question definitions.
-   */
-  CompletionStage<ImmutableMap<LifecycleStage, ImmutableList<ProgramDefinition>>> relevantPrograms(
-      long applicantId);
-
   /** Return the name of the given applicant id. */
   CompletionStage<Optional<String>> getName(long applicantId);
 
   /** Return the email of the given applicant id if they have one. */
   CompletionStage<Optional<String>> getEmail(long applicantId);
 
   /**
-   * Return all applications, including applications from previous versions, with program,
-   * applicant, and account associations eager loaded.
+   * Return all applications for all applicants, including applications from previous versions, with
+   * program, applicant, and account associations eager loaded.
    */
   ImmutableList<Application> getAllApplications();
+
+  /**
+   * Return all programs that are appropriate to serve to an applicant. Appropriate programs are
+   * those where the applicant:
+   *
+   * <ul>
+   *   <li>Has a draft application
+   *   <li>Has previously applied
+   *   <li>Any other programs that are public
+   * </ul>
+   */
+  CompletionStage<RelevantPrograms> relevantProgramsForApplicant(long applicantId);
+
+  @AutoValue
+  public abstract static class ApplicantProgramData {
+    public abstract ProgramDefinition program();
+
+    public abstract Optional<Instant> latestApplicationSubmitTime();
+
+    static ApplicantProgramData create(
+        ProgramDefinition program, Optional<Instant> latestApplicationSubmitTime) {
+      return new AutoValue_ApplicantService_ApplicantProgramData(
+          program, latestApplicationSubmitTime);
+    }
+  }
+
+  @AutoValue
+  public abstract static class RelevantPrograms {
+    public abstract ImmutableList<ApplicantProgramData> inProgress();
```

Comment:
Ah, I thought it might be permissible to only set some of these values such that we'd want default otherwise.

Curious, why have the builder rather than the default create method that requires all 3 be passed in?